model/src/main/java/com/javahelp/model/user/User.java

```diff
@@ -0,0 +1,86 @@
+package com.javahelp.model.user;
+
+/**
+ * A class for a user entity
+ */
+public class User {
+
+    /**
+     * The ID of the user account (randomly generated and immutable)
+     */
+    private String stringID;
```

Comment:
I think if this is immutable then should we make it a final attribute (e.g. private final String stringID)?

Reply:
This is only immutable in the sense that once assigned, and entered into the database, it is never changed by the user. There are still scenarios where it needs to be changed in the code, especially according to the plan in our CRC cards for creating new users. The CRC specification currently details a create method that accepts a `User` object, creates a `User` object matching the provided one in the database, assigns a random ID to the newly created user, and mutates the passed `User` object to have an ID matching its new ID in the database. To do this, we need the `stringID` field to not be final.